src/custom-elements/src/components/date-picker/date-picker.tsx

```diff
@@ -338,27 +292,78 @@ export class DatePicker {
     }, CHI_TIME_AUTO_SCROLL_DELAY);
   }
 
+  @Listen('chiDateChange')
+  handleDateChange(ev: CustomEvent) {
+    const target = ev.target as HTMLElement;
+
+    if (target.nodeName.toUpperCase() === 'CHI-DATE-PICKER') {
+      return;
+    }
```

Comment:
If this `@Listen` is removed and instead, use `<chi-date onChiDateChange={ev => this.handleDateChange(ev)}>`, this function will only be executed by the event emitted from chi-date.

Then, in this function:
```suggestion
  handleDateChange(ev: CustomEvent) {
    ev.stopPropagation();
    
    ....
```

will stop the event and fix the double event emission. Same can be done with chi-time and chiTimeChange.